### ISSUE TITLE
Object Traversal

### ISSUE THREAD

**Issue by DuncanUszkay1** [jtenner/as-pect#191]
I see that here in the [README](https://github.com/jtenner/as-pect#tostrictequal) it's mentioned that as-pect cannot perform object traversal. 

Is that because object traversal is not yet implemented or is there some reason why object traversal shouldn't be implemented in this library?

**Comment by willemneal** [comment]
Currently you can't iterate over an object's properties, however, something like [this](https://github.com/nearprotocol/near-bindgen-as) could do the trick as it adds an encode method on objects that generates a json string which could be compared.  This is expensive though, so perhaps using a similar method to just compare would be better.

**Comment by jtenner** [comment]
There is no way to reflect an object's properties without a lot of work on the compiler end. This is a limitation of the language, not the testing software.

**Comment by jtenner** [comment]
The only other reasonable alternative might be to do something like this:

```ts
class Vec3 {
  @operator("==") protected __compare(vec: Vec3 | null): bool {
   // do custom comparison here and use the `toBe()` method
  }
}
```

**Comment by DuncanUszkay1** [comment]
Thanks for letting me know 👍

**Comment by jtenner** [comment]
Reopening because I think I may have figured out how to do this given current technology.

**Comment by jtenner** [comment]
This feature has been almost completely tested in the current strict-equal-tranform branch.

I'm incredibly proud to say that it's very functional. 

Current roadblocks:

1. Documentation
2. Map key garbage collection problem (std lib?) 
3. More complex Test cases

**Comment by jtenner** [comment]
This is now live on master. The problem is that we need to release a version of as-pect so people can start testing it. I will close this issue once master is published and an alpha is released.

**Comment by jtenner** [comment]
The alpha was released and it should be okay to use with 0.9.2

It's backwards incompatible and the journey to 3.0 is going to be very long. Thank you for your patience and help.
